pkg/common/helper_test.go

```diff
@@ -353,6 +353,130 @@ func (suite *StripPrefixesTestSuite) TestPositive() {
 	suite.Require().Equal("prefix_something_1", stripped)
 }
 
+type MapStringToStringMatchByLabelSelectorTestSuite struct {
+	suite.Suite
+}
+
+func (suite *MapStringToStringMatchByLabelSelectorTestSuite) Test() {
+	for _, testCase := range []struct {
+		name                 string
+		mapStringToString    map[string]string
+		encodedLabelSelector string
+		matching             bool
+		expectedError        bool
+	}{
+		{
+			name: "SingleLabel",
+			mapStringToString: map[string]string{
+				"c": "d",
+			},
+			encodedLabelSelector: "c=d",
+			matching:             true,
+		},
+		{
+			name: "PartialLabels",
+			mapStringToString: map[string]string{
+				"a": "b",
+				"c": "d",
+			},
+			encodedLabelSelector: "a=b",
+			matching:             true,
+		},
+		{
+			name: "AllLabels",
+			mapStringToString: map[string]string{
+				"a": "b",
+				"c": "d",
+			},
+
+			encodedLabelSelector: "a=b,c=d",
+			matching:             true,
+		},
+		{
+			name: "EmptyEncodedLabel",
+			mapStringToString: map[string]string{
+				"a": "b",
+			},
+
+			encodedLabelSelector: "",
+			matching:             true,
+		},
+		{
+			name:                 "EmptyEncodedLabelEmptyLabels",
+			mapStringToString:    map[string]string{},
+			encodedLabelSelector: "",
+			matching:             true,
+		},
+		{
+			name:              "EmptyEncodedLabelNilLabels",
+			mapStringToString: nil,
+
+			encodedLabelSelector: "",
+			matching:             true,
+		},
+
+		// miss match
+		{
+			name: "EncodedLabelNotInLabels",
+			mapStringToString: map[string]string{
+				"a": "b",
+				"c": "d",
+			},
+
+			encodedLabelSelector: "z=w",
+			matching:             false,
+		},
+		{
+			name:              "EncodedLabelNilLabels",
+			mapStringToString: nil,
+
+			encodedLabelSelector: "a=b",
+			matching:             false,
+		},
+		{
+			name:                 "EncodedLabelNoLabels",
+			mapStringToString:    map[string]string{},
+			encodedLabelSelector: "z=w",
+			matching:             false,
+		},
+		{
+			name: "ValueMatchKeyNot",
+			mapStringToString: map[string]string{
+				"c": "d",
+			},
+			encodedLabelSelector: "a=d",
+			matching:             false,
+		},
+		{
+			name: "KeyMatchValueNot",
+			mapStringToString: map[string]string{
+				"c": "d",
+			},
+			encodedLabelSelector: "c=a",
+			matching:             false,
+		},
+
+		// explode
+		{
+			name:                 "InvalidEncodedLabelSelector",
+			mapStringToString:    nil,
+			encodedLabelSelector: "!@#$",
+			expectedError:        true,
+		},
```

Comment:
Why are you maintaining unit tests for `k8s.io/apimachinery` logic in nuclio?
I think `TestImportProjectSkipBySelectors` covers the `nuctl` use case nicely

Reply:
some are there for the sake of functionality, to ensure when bumping packages - no harm.
that being said, I removed half of the tests are they were some overlap themselves and some are just .. irrelevant.
`TestImportProjectSkipBySelectors` is integration test, but not cover all cases 🤷🏻‍♂️